src/lib.ts

```diff
@@ -0,0 +1,134 @@
+import { spawn } from 'child_process';
+import { copy } from 'cpx';
+import * as fs from 'fs';
+import * as globby from 'globby';
+import * as ora from 'ora';
+import * as path from 'path';
+import * as rimraf from 'rimraf';
+
+const postcss = require('postcss');
+const postcssClean = require('postcss-clean');
+const postcssCustomProperties = require('postcss-custom-properties');
+const postcssImport = require('postcss-import');
+const postcssModules = require('postcss-modules');
+const postcssPresetEnv = require('postcss-preset-env');
+
+const basePath = process.cwd();
+const srcPath = path.join(basePath, 'src');
+const assetExtensions = ['gif', 'png', 'jpg', 'jpeg', 'svg', 'eot', 'ttf', 'woff', 'woff2'];
+const assetFiles = path.join(srcPath, `**/*.{${assetExtensions.join(',')}}`);
+
+function createChildProcess(command: string, args: string[], errorMessage: string) {
+	return new Promise((resolve, reject) => {
+		const child = spawn(path.join(basePath, 'node_modules/.bin', command), args, { cwd: basePath });
+		child.on('error', reject);
+		child.on('exit', code => (code !== 0 ? reject(new Error(errorMessage)) : resolve()));
+	});
+}
+
+function createTask(callback: any) {
+	return new Promise((resolve, reject) => {
+		callback((error?: Error) => {
+			if (error) {
+				reject(error);
+			}
+			resolve();
+		});
+	});
+}
+
+function wrapUmd(content: string): string {
+	return `(function (root, factory) {
+if (typeof define === 'function' && define.amd) {
+	define([], function () { return (factory()); });
+} else if (typeof module === 'object' && module.exports) {
+	module.exports = factory();
+}
+}(this, function () {
+	return ${content};
+}));`;
+}
+
+function buildCss({ mode = 'dist', legacy }: any) {
+	const outDir = `output/${mode}`;
+	const cssFiles = path.join(basePath, outDir, '**/*.css');
+	const packageJsonPath = path.join(basePath, 'package.json');
+	const packageJson = fs.existsSync(packageJsonPath) ? require(packageJsonPath) : {};
+	const postcssPresetConfig = {
+		browsers: legacy ? ['last 2 versions', 'ie >= 10'] : ['last 2 versions'],
+		features: {
+			'nesting-rules': true
+		},
+		autoprefixer: {
+			grid: legacy
+		}
+	};
+
+	const modulesProcessor = postcss([
+		postcssImport(),
+		postcssPresetEnv(postcssPresetConfig),
+		postcssModules({
+			generateScopedName: mode === 'dist' ? '[hash:base64:8]' : '[name]__[local]__[hash:base64:5]',
```

Comment:
We have standardised on just `'[name]__[local]__[hash:base64:5]'` now for all modes for the other build commands

Reply:
Thanks, I should have looked at what we were already using in the repo instead of what `@dojo/widgets` was using from `@dojo/grunt-dojo2`.